### DISCUSSION TITLE
Give initial route constraint

### DISCUSSION THREAD

**Discussion by despondency** [google/or-tools#2768] [Routing (and legacy CP) questions] [answered]
Hello, I have a question regarding the API, since i can't really find any example/solution to this:

If we have nodes A B C D E F G H -> and we want to do VRPTW on them. However we want to have specifically D E G in their specific order (D < E < G) in terms of position in the path

We want to be able to construct a path, by making some of the nodes keep their relative position to each other, while other nodes can be in any place in the path sequence, Is that possible?

Whole example :

A B C [D] [E] F [G] H -> we want D < E < G

We can have such orders:

B A C [D] F [E] [G] H
B A C F H [D] [E] [G] 
B [D] A C F [E] [G] H
B C A [D] [E] F [G] H

and so on.

**Comment by Mizux** [comment, marked as answer]
Already answered on or-tools-discuss...

Basically you need a strictly increasing monotonic dimension (e.g. distance RoutingModelDimension, or a counter Dimension), then you can add few constraints...
```python
# constant transit dimension, i.e. each visit add +1 to the CumulVar
routing.AddConstantDimension(1, 100, True, 'Counter')
counter_dim = routing.GetDimensionOrDie('Counter')

solver = routing.solver()
d_index = manager.NodeToIndex(d_node)
e_index = manager.NodeToIndex(e_node)
g_index = manager.NodeToIndex(g_node)
solver.Add(counter_dim.CumulVar(d_index) < counter_dim.CumulVar(e_index))
solver.Add(counter_dim.CumulVar(e_index) < counter_dim.CumulVar(g_index))

```

**Comment by despondency** [comment]
Much thanks @Mizux you are great!
